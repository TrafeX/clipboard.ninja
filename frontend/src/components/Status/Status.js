import React, { useState } from 'react';
import Snackbar from '@material-ui/core/Snackbar';

const Status = ({status}) => {

  const [lastMessage, setLastMessage] = useState('');

  return (
    <Snackbar
      open={status !== lastMessage}
      onClose={() => setLastMessage(status)}
      message={status}
      autoHideDuration={4000}
    />
  );
};

export default Status;
