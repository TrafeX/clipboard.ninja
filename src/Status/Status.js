// @flow
import React, { Component } from 'react';
import Snackbar from 'material-ui/Snackbar';

type State = State;
type Props = {
  status: string
};

class Status extends Component<void, Props, State> {

  shouldComponentUpdate = (nextProps: Props) => {
    return nextProps.status !== '' && nextProps.status !== this.props.status;
  };

  render() {
    if (this.props.status === '') {
      return null;
    }
    return (
      <Snackbar
        open={true}
        message={this.props.status}
        autoHideDuration={4000}
      />
    );
  }
}

export default Status;
