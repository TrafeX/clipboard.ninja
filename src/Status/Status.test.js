import React from 'react';
import Status from './Status';
import { shallow } from 'enzyme';

it('renders without crashing', () => {
  const wrapper = shallow(<Status status={'Testing'}/>)
  expect(wrapper).toMatchSnapshot();
});
