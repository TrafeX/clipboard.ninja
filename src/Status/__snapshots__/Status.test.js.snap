// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Status
    status="Testing"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "autoHideDuration": 4000,
      "message": "Testing",
      "open": true,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "autoHideDuration": 4000,
        "message": "Testing",
        "open": true,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
